airflow/models/dag.py

```diff
@@ -767,7 +767,7 @@ def __init__(
 
         self.doc_md = self.get_doc_md(doc_md)
 
-        self.tags = tags or []
+        self.tags: abc.MutableSet[str] = set(tags or [])
```

Comment:
This should probably just use `Collection`. We don’t really expect users to add more tags to a DAG after it’s created.

Reply:
I don't think it's true. In [cluster policies](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/cluster-policies.html), we may want to add custome tags to DAGs (for orginization hierarcy for example).
And I don't think it would be correct to make it an immutable collection that we would need to create every time we want to change something.

If we would go with `Collection`, we would go with a different programing paradigram, and I am not sure everyone would like it, so I would keep the door open and let everyone chose what they want to.